lib/teleterm/apiserver/apiserver.go

```diff
@@ -119,68 +126,3 @@ type APIServer struct {
 	// grpc is an instance of grpc server
 	grpcServer *grpc.Server
 }
-
-func getGrpcCredentials(cfg Config) (grpc.ServerOption, error) {
-	uri, err := utils.ParseAddr(cfg.HostAddr)
-
-	if err != nil {
-		return nil, trace.BadParameter("invalid host address: %s", cfg.HostAddr)
-	}
-
-	if uri.Network() != "unix" {
-		keyPair, err := generateKeyPair(cfg.CertsDir)
-		if err != nil {
-			return nil, trace.Wrap(err)
-		}
-
-		return grpc.Creds(keyPair), nil
-	}
-
-	return grpc.Creds(nil), nil
-}
-
-func generateKeyPair(certsDir string) (credentials.TransportCredentials, error) {
```

Comment:
This function used to do two things. I extracted them into separate functions.

* Create a key pair for the tshd process and save the public key on disk.
   * This is now `generateAndSaveCert`.
* Create gRPC server credentials.
   * This is now `createServerCredentials`.